### DISCUSSION TITLE
Will FutureProofHomes Satellite1 hardware and firmware work with 48kHz sample rate (or better) for audio output to speaker and audio-jack?

### DISCUSSION THREAD

**Discussion by Hedda** [FutureProofHomes/Satellite1-Hardware#6] [General]
Will FutureProofHomes Satellite1 hardware and firmware work with 48kHz sample rate (or better) for audio output to speaker and audio-jack?

@FutureProofHomes @that1guy @alextrical Asking since just read a comment from formatBCE in a ReSpeaker Lite discussion thread in Home Assistant's community forum as an answer to if it is beneficial to use two i2s modules where he wrote that the FutureProofHomes Satellite1 hardware design uses use a single duplex i2s channel for both microphone and speaker which might be limited it if comparing to the upcoming Home Assistant Voice Satellite hardware from Nabu Casa that will use two separate i2s channels for mics and speakers.

https://community.home-assistant.io/t/respeaker-lite-new-seeed-studio-voice-assistant-development-kit-hardware-combine-esp32-with-xmos-xu316-dsp-chip-for-advanced-audio-processing-as-a-esphome-based-home-assistant-assist-satellite-voice-devkit/756944/414

Quote: 

>> _Are there any new findings concerning the usage of two I2S modules in the XMOS firmware of Nabu Casa instead of using only one as it seems to be the case with the default respeaker lite I2S firmware? Is it beneficial to use two I2S modules?_
>>
> _This is guided by hardware difference. Nabu PE device has separate i2s channels for mics/speaker. Respeaker Lite de s (and FPH SAT1 devs too) decided to use duplex I2S for both. Similar pinout is used in ESP32-S3-BOX._
>
> _**Single I2S saves GPIO pins. But it has limitation: you can’t use different sample rate for speaker and microphone. It’s either 16kHz, or 48kHz for both.**_
> 
> _**16kHz is enough for speech, so Nabu devs made all the parts of input pipeline work on 16kHz. But for decent sound 48kHz are needed on speaker. So Nabu code is using 48kHz for output. In their case it’s easy, because they have separate I2S and can assign different sample rates.**_
> 
> _But for Respeaker Lite there was a tough time. First we had 16kHz firmware exclusively. It was fine, but the sound was not so good and unacceptable for music playback. Then I asked guys from Seeed for 48kHz firmware. Eventually they delivered it. But microphone wasn’t working, because MWW and VA components of Nabu code are expecting 16kHz._
> 
> _Then Respeaker community made fork for nabu_microphone repo, and reduced sample rate there with simple resampling. So now my code is using that fork._
> 
> _In conclusion, we have difference between PE and Respeaker, but both working good. :)_

**Comment by formatBCE** [comment]
As you may see in https://github.com/FutureProofHomes/Satellite1-ESPHome/blob/ld2410_settings_template/config/common/voice_assistant.yaml#L27 and https://github.com/FutureProofHomes/Satellite1-ESPHome/blob/ld2410_settings_template/config/common/media_player.yaml#L29, it's 48kHz. I believe the microphone is downsampled to 16kHz, similar to Respeaker. No worries, sound will be decent! :)

**Comment by that1guy** [comment]
Correct.  We have 48kHz output to both the headphone and amplified speaker DAC.  Mics are being processed at 16kHz.

@formatBCE is a private beta testing the hardware now, so he knows what he is talking about. :)

@Hedda please feel free to close this conversation if we've answered the question.   Cheers!
